### PR TITLE
Update prometheus/alertmanager to version v0.25.1-0.20230505130626-263ca5c9438e

### PR DESCRIPTION
**Pull request by krishnateja325**
**What this PR does**:
Update Prometheus to [prometheus/alertmanager@263ca5c](https://github.com/prometheus/alertmanager/commit/263ca5c9438e3b2e2af95a2f1e4f53d32ab18e5c)

This includes 
- [prometheus/alertmanager/#3094](https://github.com/prometheus/alertmanager/pull/3094) for filtering 5xx Errors in numTotalFailedNotifications metric.
- [prometheus/alertmanager/#3352](https://github.com/prometheus/alertmanager/pull/3352) Delete silence respond with 404 when silence is not found
- [prometheus/alertmanager/#3220](https://github.com/prometheus/alertmanager/pull/3220) Refactor nflog configuration options to make it similar to Silences.
- [prometheus/alertmanager/#3228](https://github.com/prometheus/alertmanager/pull/3228) mark webhook URL as a secret
- [prometheus/alertmanager/#3209](https://github.com/prometheus/alertmanager/pull/3209) Changes receiver and inhibition rules arrays to slices of value items instead of pointer items
- Validate the Webhook config to block the setting `url_file` that got introduced as part of [prometheus/alertmanager@41eb121](https://github.com/prometheus/alertmanager/commit/41eb1213bb1c7ce0aa9e6464e297976d9c81cfe5)

Latest [released version](https://github.com/prometheus/alertmanager/releases/tag/v0.25.0) only contains changes till 2022-12-22. Hence updating the prometheus/alertmanger to include this change. 
**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Tests updated
- [NA] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by alvinlin123** [comment]
I think it would be useful if we can pull in the prometheus/alertmanager commit that includes https://github.com/prometheus/alertmanager/pull/3352 as part of this PR too.

**Comment by krishnateja325** [comment]
Sure, I can try including that as well @alvinlin123

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## master / unreleased
+* [CHANGE] Alertmanager: Validating new fields on the Webhook AM config. #5276
```
Comment (yeya24) [inline CHANGELOG.md:4]: Since this update includes multiple changes from AM and the validation change, shall we mention more update/changes we get?

**Review by yeya24 — APPROVED** [review]
Thanks. I think it is really a great job! I have only small nit about the changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # Changelog
 
 ## master / unreleased
+* [CHANGE] Updating prometheus/alertmanager from v0.25.0 to v0.25.1-0.20230505130626-263ca5c9438e. This includes the below changes. #5276
```
Comment (krishnateja325) [inline CHANGELOG.md:4]: Included user facing changes in the changelog.

File: CHANGELOG.md (hunk 1 above)
Comment (krishnateja325) [reply CHANGELOG.md]: sure will include more changes in the change log.

File: vendor/github.com/prometheus/alertmanager/config/notifiers.go [hunk 3]
```diff
@@ -686,11 +696,17 @@ func (c *PushoverConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
...
-		return fmt.Errorf("missing user key in Pushover config")
+	if c.UserKey == "" && c.UserKeyFile == "" {
+		return fmt.Errorf("one of user_key or user_key_file must be configured")
 	}
-	if c.Token == "" {
-		return fmt.Errorf("missing token in Pushover config")
+	if c.UserKey != "" && c.UserKeyFile != "" {
+		return fmt.Errorf("at most one of user_key & user_key_file must be configured")
+	}
+	if c.Token == "" && c.TokenFile == "" {
```
Comment (alvinlin123) [inline vendor/github.com/prometheus/alertmanager/config/notifiers.go:705]: We need to disallow `TokenFIle` for PushOver also.

File: vendor/github.com/prometheus/alertmanager/config/notifiers.go [hunk 4]
```diff
@@ -686,11 +696,17 @@ func (c *PushoverConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 	if err := unmarshal((*plain)(c)); err != nil {
 		return err
 	}
-	if c.UserKey == "" {
-		return fmt.Errorf("missing user key in Pushover config")
+	if c.UserKey == "" && c.UserKeyFile == "" {
```
[the remark this answers is not in the document]
Comment (krishnateja325) [reply vendor/github.com/prometheus/alertmanager/config/notifiers.go:699]: Added validation to disallow `UserKeyFile` in PushOver Config.

File: vendor/github.com/prometheus/alertmanager/config/notifiers.go (hunk 3 above)
Comment (krishnateja325) [reply vendor/github.com/prometheus/alertmanager/config/notifiers.go:705]: Added validation to disallow `TokenFile` in PushOver Config.

**Review by alvinlin123 — APPROVED** [review]

**Pull request merged**